include/swift/AST/DiagnosticsSema.def

```diff
@@ -2337,6 +2337,9 @@ WARNING(override_swift3_objc_inference,none,
         "override of %0 %1 from extension of %2 depends on deprecated "
         "inference of '@objc'",
         (DescriptiveDeclKind, DeclName, Identifier))
+ERROR(override_method_different_generic_sig,none,
+      "cannot override method %0 with a different generic signature",
+      (DeclBaseName))
```

Comment:
This diagnostic would be better if it actually contained the generic signatures printed out as StringRefs. Eg, "override method''s generic signature %0 is not compatible with base method generic signature %1"